generator/src/main/java/com/scottlogic/deg/generator/walker/reductive/fieldselectionstrategy/FieldAppearanceFixingStrategy.java

```diff
@@ -47,7 +47,13 @@ public Field getNextFieldToFix(ReductiveState reductiveState) {
     }
 
     private Comparator<Map.Entry<Field, Integer>> highestToLowest() {
-        return Collections.reverseOrder(Map.Entry.comparingByValue());
+        return Collections.reverseOrder(this::compare);
     }
 
+    private int compare(Map.Entry<Field, Integer> a, Map.Entry<Field, Integer> b) {
+        if (a.getValue() != b.getValue()) {
```

Comment:
`a.getValue() != b.getValue()` is checking reference inequality, not value inequality.